### PR TITLE
full tests for dectree config validate

### PR DESCRIPTION
**Pull request by handwerkerd**
Changes proposed in this pull request:

- I filled out the tests for validate_tree. @jbteves Let me know if this looks reasonable to you.
-

### CODE REVIEW

**Comment by jbteves** [comment]
This looks good, but what if we did dictionaries specified in something like `dtm_test_dicts.py`? Then you could have comments (which aren't something available in JSON without some hackiness).

**Comment by handwerkerd** [comment]
I tried that, but I realized that file could get really ugly with a bunch of different trees stacked on top of each other. Even the minimal tree is a bit long and includes a list of dictionaries inside a dictionary field. I ended up using the info field in each tree to explain what is good or bad about it. 

Once nice side-effect of this approach is that I can make sure each default tree is valid, so that, if the validator is changed to reject a default tree, we'll know.

Worth changing to `dtm_test_dicts.py` or keep as is?

**Comment by jbteves** [comment]
The info field helps but I feel it would be a little more clear to directly comment on the "weird" parameter in the tree.
You could make a "correct" tree return from a function, and then have other functions ingest that weird tree and then modify it to make it weird, which wouldn't be too complex.

**Comment by jbteves** [comment]
I would separate testing for packaged trees from validation of a tree more generally.

**Comment by handwerkerd** [comment]
Since the goal of this is partially to get me thinking about tests correctly, I'll give it a try.
That said, I do think `test_validate_tree_succeeds` should run on all valid default trees. If someone messes with a default tree or messes with validation criteria, we'd want to automatically see if that makes any of the default trees fail validation.

**Comment by jbteves** [comment]
You could make an additional test, `test_default_trees_succeed` to separate general validity from specific tree validity.

**Comment by handwerkerd** [comment]
Now using a dict in the same file. I also changed the testing file to test_ComponentSelector.py

Successful tests. They helped me notice & fix a few bugs in the code.
- A typo in a variable name in a warning is fixed
- Now looking for an undefined required parameter name in addition to undefined optional parameter names
- In minimal tree, an optional parameter is now defined with optional parameters rather than required parameters
- Fixed an incorrect default parameter option in selection.nodes.dec_variance_lessthan_thresholds

I hope this isn't a sign of things to come as I add more tests.

**Comment by jbteves** [comment]
It probably is, this is why I recommend writing tests first when reasonable. But after is better than never by wide margins. Would you like to leave this open to add more tests or have me merge this?

File: tedana/tests/test_ComponentSelector.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+        "missing_key": A dict missing one of the required keys (refs)
+
+    Returns
+    -------
+    tree: :ojb:`dict` A dict that can be input into ComponentSelector.validate_tree
+    """
+
+    # valid_dict is a simple valid dictionary to test
+    # It includes a few things that should trigger warnings, but not errors:
+    #  One unused key: "unused_key"
```
Comment (jbteves) [inline tedana/tests/test_ComponentSelector.py:36]: I would highlight these changes inline as well, all at the top is harder to read.

**Review by jbteves — APPROVED** [review]

**Pull request merged**